packages/x-date-pickers/src/internals/components/PickersProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { PickerValidDate } from '../../models';
+import { PickersInputLocaleText } from '../../locales';
+import { LocalizationProvider } from '../../LocalizationProvider';
+
+export const PickersFieldContext = React.createContext<PickersFieldContextValue | null>(null);
+
+/**
+ * Provides the context for the various parts of a picker component:
+ * - fieldContextValue: the context for the field
+ * - localizationProvider: the translations passed through the props and through a parent LocalizationProvider.
+ *
+ * @ignore - do not document.
+ */
+export function PickersProvider<TDate extends PickerValidDate>(
+  props: PickersFieldProviderProps<TDate>,
+) {
+  const { fieldContextValue, localeText, children } = props;
+
+  return (
+    <PickersFieldContext.Provider value={fieldContextValue}>
+      <LocalizationProvider localeText={localeText}>{children}</LocalizationProvider>
+    </PickersFieldContext.Provider>
+  );
+}
+
+interface PickersFieldProviderProps<TDate extends PickerValidDate> {
+  fieldContextValue: PickersFieldContextValue;
+  localeText: PickersInputLocaleText<TDate> | undefined;
+  children: React.ReactNode;
+}
+
+export interface PickersFieldContextValue {
```

Comment:
We need to discuss the exact DX here:

1. Should the behavior be `onOpen` or `onToggle`?
2. Should it be named `onOpen` or `onOpenViews` or `onOpenParent` or something else?

Right now it's not used by the built-in fields, but it will allow to make this demo and the button demo a lot simpler.
And as discussed last week, I would love to migrate to this approach for the built-in fields as well at some point, to make the picker more agnostic.

Reply:
1. Since it's handling not only the opening but also the closing of the views, I would say `onToggle` seems more precise
2. I personally prefer having the suffix "Views" to explicit which component it's related to, even though there isn't multiple openable components to be differentiated

Reply:
> Since it's handling not only the opening but also the closing of the views, I would say onToggle seems more precise


Most of the time it's not the field that closes the view, but the view itself (or more precisely the popper wrapping it).
The Popper detects stuff like click outside, press on `Escape` etc... and closes itself.


But once we have a fully unstyled picker and people can wrap their view in whatever UI they decide, we could have some use cases where the closing is done by the field (imagine the view being inside some sort of `Collapsible` for example).
So I tend to agree with you that `onToggle` would be more versatile here.

Reply:
Renamed to `onToggleView` and added some JSDoc